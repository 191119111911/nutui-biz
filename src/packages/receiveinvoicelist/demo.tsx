import React from 'react'
import { ReceiveInvoiceItem, ReceiveInvoiceList } from './receiveinvoicelist'
import { useTranslate } from '../../sites/assets/locale'
import { Toast } from '@nutui/nutui-react';

interface tarnslatedOption {
  basic: string;
  useSwipe: string;
}

const ReceiveInvoiceListDemo = () => {
  const [translated] = useTranslate<tarnslatedOption>({
    'zh-CN': {
      basic: '基本用法',
      useSwipe: '使用左滑删除',
    },
    'en-US': {
      basic: 'Basic Usage',
      useSwipe: 'Use Swipe Delete',
    }
  });

  const state = {
    modelValue: 1,
    list: [
      {
        id: 1,
        name: '张三',
        tel: '15088888888',
        addres: '北京市大兴京东大厦1号楼',
        isDefault: true,
      },
      {
        id: 2,
        name: '李四',
        tel: '15088888888',
        addres: '北京市大兴京东大厦2号楼',
        isDefault: false,
        extends: [
          { label: '扩展1', value: '扩展信息展示' },
          { label: '扩展2', value: '扩展信息展示' }
        ]
      }
    ]
  };

  const event = {
    onEdit: (item: ReceiveInvoiceItem, index: number) => {
      Toast.text('onEdit ' + item.name);
      console.log('onEdit', item, index);
    },
    onSelected: (item: ReceiveInvoiceItem, index: number) => {
      Toast.text('onSelected ' + item.name);
      console.log('onSelected', item, index);
    },
    onDelete: (item: ReceiveInvoiceItem, index: number) => {
      Toast.text('onDelete ' + item.name);
      console.log('onDelete', item, index);
    }
  }


  return (
    <>
      <div className="demo">
        <h2>{translated.basic}</h2>
        <ReceiveInvoiceList list={state.list} modelValue={state.modelValue} onSelected={event.onSelected} onEdit={event.onEdit} />
        <h2>{translated.useSwipe}</h2>
        <ReceiveInvoiceList enableDelete={true} list={state.list} modelValue={state.modelValue} onSelected={event.onSelected} onEdit={event.onEdit} onDelete={event.onDelete} />
      </div>
    </>
  )
}

export default ReceiveInvoiceListDemo
