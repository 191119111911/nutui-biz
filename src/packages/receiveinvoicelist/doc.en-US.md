#  ReceiveInvoiceList

### Intro

Commonly used to display contact list information, etc.

### Install

```javascript
import { ReceiveInvoiceList } from '@nutui/nutui-biz';
```

## Demo

### Basic Usage

:::demo

```ts
import  React from 'react';
import { ReceiveInvoiceList } from '@nutui/nutui-biz';
import { Toast } from '@nutui/nutui-react';

export interface ReceiveInvoiceItemExt {
  label: string;
  value: string;
  [x: string]: any;
}
export interface ReceiveInvoiceItem {
  id: number | string;
  name: string;
  tel: string;
  addres: string;
  isDefault: boolean;
  extends?: Array<ReceiveInvoiceItemExt>;
  [x: string]: any;
}

const App = () => {
  
  const state = {
    defaultValue: 1,
    list: [
      {
        id: 1,
        name: '张三',
        tel: '15088888888',
        addres: '北京市大兴京东大厦1号楼',
        isDefault: true,
      },
      {
        id: 2,
        name: '李四',
        tel: '15088888888',
        addres: '北京市大兴京东大厦2号楼',
        isDefault: false,
        extends: [
          { label: '扩展1', value: '扩展信息展示' },
          { label: '扩展2', value: '扩展信息展示' }
        ]
      }
    ]
  };

  const event = {
    onEdit: (item: ReceiveInvoiceItem,index:number) => { 
      Toast.text('onEdit ' + item.name);
      console.log('onEdit', item,index);
    },
    onSelected: (item: ReceiveInvoiceItem,index:number) => { 
      Toast.text('onSelected ' + item.name);
      console.log('onSelected', item,index) 
    }
  }

  return (
     <ReceiveInvoiceList 
      list={state.list} 
      defaultValue={state.defaultValue} 
      onSelected={event.onSelected} 
      onEdit={event.onEdit} 
     />
  );
};
export default App;
```

:::
### Use Swipe Delete

:::demo

```ts
import  React from 'react';
import { ReceiveInvoiceList } from '@nutui/nutui-biz';
import { Toast } from '@nutui/nutui-react';

export interface ReceiveInvoiceItemExt {
  label: string;
  value: string;
  [x: string]: any;
}
export interface ReceiveInvoiceItem {
  id: number | string;
  name: string;
  tel: string;
  addres: string;
  isDefault: boolean;
  extends?: Array<ReceiveInvoiceItemExt>;
  [x: string]: any;
}

const App = () => {
  
  const state = {
    defaultValue: 1,
    list: [
      {
        id: 1,
        name: '张三',
        tel: '15088888888',
        addres: '北京市大兴京东大厦1号楼',
        isDefault: true,
      },
      {
        id: 2,
        name: '李四',
        tel: '15088888888',
        addres: '北京市大兴京东大厦2号楼',
        isDefault: false,
        extends: [
          { label: '扩展1', value: '扩展信息展示' },
          { label: '扩展2', value: '扩展信息展示' }
        ]
      }
    ]
  };

  const event = {
    onEdit: (item: ReceiveInvoiceItem,index:number) => { 
      Toast.text('onEdit ' + item.name);
      console.log('onEdit', item,index);
    },
    onSelected: (item: ReceiveInvoiceItem,index:number) => { 
      Toast.text('onSelected ' + item.name);
      console.log('onSelected', item,index);
    },
    onDelete: (item: ReceiveInvoiceItem,index:number) => { 
      Toast.text('onDelete ' + item.name);
      console.log('onDelete', item,index);
    },
  }

  return (
     <ReceiveInvoiceList 
      enableDelete={true} 
      list={state.list} 
      defaultValue={state.defaultValue} 
      onSelected={event.onSelected} 
      onEdit={event.onEdit} 
      onDelete={event.onDelete} 
     />
  );
};
export default App;
```

:::

## API

### Props


| Attribute    | Description                           | Type                      | Default |
|--------------|---------------------------------------|---------------------------|---------|
| defaultValue | Id of chosen contact                  | number \| string          | ""      |
| list         | Data List                             | Array<ReceiveInvoiceItem> | []      |
| enableDelete | Whether to enable the delete function | boolean                   | `false` |
| customEdit   | Custom Edit Button                    | ReactNode                 |         |

### ReceiveInvoiceItem Data Structure

| key       | Description               | Type                         |
|-----------|---------------------------|------------------------------|
| id        | ID                        | number \| string             |
| name      | Name                      | string                       |
| tel       | Phone                     | string                       |
| addres    | Addres                    | string                       |
| isDefault | Is it the default address | boolean                      |
| extends   | Extend custom array       | Array<ReceiveInvoiceItemExt> |
### ReceiveInvoiceItemExt Data Structure

| key   | Description        | Type   |
|-------|--------------------|--------|
| label | Custom field name  | string |
| value | Custom field value | string |


### Events
| Attribute  | Description    | Arguments                              |
|------------|----------------|----------------------------------------|
| onEdit     | Edit Event     | item\:ReceiveInvoiceItem,index\:number |
| onSelected | Selected Event | item\:ReceiveInvoiceItem,index\:number |
| onDelete   | Delete Event   | item\:ReceiveInvoiceItem,index\:number |
