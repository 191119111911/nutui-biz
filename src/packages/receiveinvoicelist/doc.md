#  ReceiveInvoiceList 收票人列表

### 介绍

常见于展示联系人列表信息等。

### 安装

```javascript
import { ReceiveInvoiceList } from '@nutui/nutui-biz';
```

## 代码演示

### 基本用法

:::demo

```ts
import  React from 'react';
import { ReceiveInvoiceList } from '@nutui/nutui-biz';

const App = () => {
  
  const state = {
    defaultValue: 1,
    list: [
      {
        id: 1,
        name: '张三',
        tel: '15088888888',
        addres: '北京市大兴京东大厦1号楼',
        isDefault: true,
      },
      {
        id: 2,
        name: '李四',
        tel: '15088888888',
        addres: '北京市大兴京东大厦2号楼',
        isDefault: false,
        extends: [
          { label: '扩展1', value: '扩展信息展示' },
          { label: '扩展2', value: '扩展信息展示' }
        ]
      }
    ]
  };

  const event = {
    onEdit: (item: ReceiveInvoiceItem) => { console.log('onEdit', item) },
    onSelected: (item: ReceiveInvoiceItem) => { console.log('onSelected', item) }
  }

  return (
     <ReceiveInvoiceList list={state.list} defaultValue={state.defaultValue} onSelected={event.onSelected} onEdit={event.onEdit} />
  );
};
export default App;
```

:::
### 使用左滑删除

:::demo

```ts
import  React from 'react';
import { ReceiveInvoiceList } from '@nutui/nutui-biz';

const App = () => {
  
  const state = {
    defaultValue: 1,
    list: [
      {
        id: 1,
        name: '张三',
        tel: '15088888888',
        addres: '北京市大兴京东大厦1号楼',
        isDefault: true,
      },
      {
        id: 2,
        name: '李四',
        tel: '15088888888',
        addres: '北京市大兴京东大厦2号楼',
        isDefault: false,
        extends: [
          { label: '扩展1', value: '扩展信息展示' },
          { label: '扩展2', value: '扩展信息展示' }
        ]
      }
    ]
  };

  const event = {
    onDelete: (item: ReceiveInvoiceItem) => { console.log('onDelete', item) },
  }

  return (
     <ReceiveInvoiceList enableDelete={true} list={state.list} defaultValue={state.defaultValue} onDelete={event.onDelete} />
  );
};
export default App;
```

:::

## API

### Props


| 字段         | 说明                | 类型                      | 默认值  |
|--------------|---------------------|---------------------------|---------|
| defaultValue | 当前选中联系人的 id | number \| string          | -       |
| list         | 联系人列表          | Array<ReceiveInvoiceItem> | []      |
| enableDelete | 是否启用删除功能    | boolean                   | `false` |

### ReceiveInvoiceItem 数据结构

| 键名      | 说明           | 类型                         |
|-----------|----------------|------------------------------|
| id        | 联系人的 id    | number \| string             |
| name      | 联系人姓名     | string                       |
| tel       | 联系人手机号   | string                       |
| addres    | 联系人地址信息 | string                       |
| isDefault | 是否为默认地址 | boolean                      |
| extends   | 扩展自定义数组 | Array<ReceiveInvoiceItemExt> |
### ReceiveInvoiceItemExt 数据结构

| 键名  | 说明           | 类型   |
|-------|----------------|--------|
| label | 自定义字段名称 | string |
| value | 自定义值       | string |


### Events
| 字段       | 说明     | 回调参数                               |
|------------|----------|----------------------------------------|
| onEdit     | 编辑事件 | item\:ReceiveInvoiceItem,index\:number |
| onSelected | 选中事件 | item\:ReceiveInvoiceItem,index\:number |
| onDelete   | 删除事件 | item\:ReceiveInvoiceItem,index\:number |
