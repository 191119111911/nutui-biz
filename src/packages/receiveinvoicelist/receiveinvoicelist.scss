.nb-receive-invoice-list {
    &__item {
        &-header {
            font-size: 14px;
            font-weight: 600;
            color: rgba(51, 51, 51, 1);
            .nut-cell__title {
                overflow: hidden;
            }
            &-title {
                display: flex;
                align-items: center;
                .nut-tag--danger {
                    flex-shrink: 0;
                }
                > p {
                    overflow: hidden;
                    text-overflow: ellipsis;
                    white-space: nowrap;
                }
            }
        }

        &-footer {
            display: flex;
            &.infobox {
                flex: 1;
                display: flex;
                flex-direction: column;
            }
            &-info {
                display: flex;
                align-items: flex-start;
                margin-bottom: 10px;
                &:last-child {
                    margin-bottom: 0;
                }
                span {
                    width: 44px;
                    font-size: 12px;
                    font-weight: normal;
                    color: #999999;
                }
                p {
                    flex: 1;
                    font-size: 12px;
                    color: #333333;
                    word-break: break-word;
                }
            }
        }
    }
}
