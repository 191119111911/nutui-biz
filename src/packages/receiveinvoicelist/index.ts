import { ReceiveInvoiceList } from './receiveinvoicelist'

export default ReceiveInvoiceList
