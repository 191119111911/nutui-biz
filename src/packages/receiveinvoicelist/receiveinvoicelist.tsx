import React, { Children, FunctionComponent, ReactNode, useState } from 'react'
import { useConfig } from '@/packages/configprovider'
import { Checkbox, Icon, Tag, Cell, CellGroup, Swipe, Button } from '@nutui/nutui-react';
import { IComponent } from '@/utils/typings'
import bem from '@/utils/bem'
import classNames from 'classnames';
import { numericProp } from '@/utils/props';
export interface ReceiveInvoiceItemExt {
  label: string;
  value: string;
  [x: string]: any;
}
export interface ReceiveInvoiceItem {
  id: numericProp;
  name: string;
  tel: string;
  addres: string;
  isDefault: boolean;
  extends?: Array<ReceiveInvoiceItemExt>;
  [x: string]: any;
}
export interface ReceiveInvoiceListProps extends IComponent {
  defaultValue: numericProp;
  list: Array<ReceiveInvoiceItem>;
  enableDelete: boolean;
  customEdit?: ReactNode;
  onEdit?: (item: ReceiveInvoiceItem, index: number) => void;
  onSelected?: (item: ReceiveInvoiceItem, index: number) => void;
  onDelete?: (item: ReceiveInvoiceItem, index: number) => void;
}
const defaultProps = {
  defaultValue: "",
  list: new Array<ReceiveInvoiceItem>,
  enableDelete: false
} as ReceiveInvoiceListProps;
export const ReceiveInvoiceList: FunctionComponent<Partial<ReceiveInvoiceListProps>> = (props) => {
  const { locale } = useConfig();
  const [defaultValue, setDefaultValue] = useState(props.defaultValue);
  const b = bem('receive-invoice-list');
  const {
    style,
    className,
    list,
    onSelected,
    onEdit,
    onDelete,
    enableDelete,
    customEdit,
    ...rest
  } = {
    ...defaultProps,
    ...props,
  }
  const onSelect = (item: ReceiveInvoiceItem, index: number) => {
    if (item.id == defaultValue) {
      setDefaultValue("")
      return
    }
    setDefaultValue(item.id);
    onSelected?.(item, index);
  }
  function RenderRowInfo(label: string, value: string) {
    return (
      <div className={b('item-footer-info')} key={label}>
        <span>{label}</span>
        <p>{value}</p>
      </div>
    );
  }
  const RenderItem = (item: ReceiveInvoiceItem, index: number) => {
    return (<CellGroup className={b('item')}>
      <Cell className={b('item-header')}
        onClick={() => onSelect(item, index)}
        title={
          <div className={b('item-header-title')}>{item.isDefault && <><Tag type="danger">{locale.default}</Tag>&nbsp;</>}<p>{item.name}</p></div>
        }
        linkSlot={
          <div className={b('item-header-edit')} onClick={(e) => { e.stopPropagation(); onEdit?.(item, index) }}>
            {customEdit ? customEdit : <Icon name="edit" color='#666' ></Icon>}
          </div>
        } />
      <Cell className={b('item-footer')} onClick={() => onSelect(item, index)}>
        <Checkbox textPosition="right" label="" checked={item.id == defaultValue} />
        <div className={b('item-footer infobox')}>
          {RenderRowInfo(locale.tel, item.tel)}
          {RenderRowInfo(locale.addres, item.addres)}
          {item?.extends?.map((i) => {
            return RenderRowInfo(i.label, i.value);
          })}
        </div>
      </Cell>
    </CellGroup>);
  }
  return (
    <div className={classNames([b(), className])} style={style} {...rest}>
      {list?.map((item, index) => {
        return (
          <React.Fragment key={item.id.toString()}>
            {enableDelete ?
              <Swipe rightAction={
                <Button type="primary" shape="square" onClick={() => onDelete?.(item, index)}>{locale.swipeShell.delete}</Button>
              }>
                {RenderItem(item, index)}
              </Swipe>
              : RenderItem(item, index)}
          </React.Fragment>
        )
      })}
    </div>
  )
}
ReceiveInvoiceList.defaultProps = defaultProps;
ReceiveInvoiceList.displayName = 'NutReceiveInvoiceList'
