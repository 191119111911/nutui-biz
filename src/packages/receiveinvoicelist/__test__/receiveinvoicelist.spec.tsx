import React from 'react'
import { fireEvent, render, waitFor } from '@testing-library/react'
import '@testing-library/jest-dom'
import { ReceiveInvoiceList } from '../receiveinvoicelist'


test('test props && event', async () => {
  const state = {
    defaultValue: 2,
    list: [
      {
        id: 1,
        name: '张三',
        tel: '15088888888',
        addres: '北京市大兴京东大厦1号楼',
        isDefault: true,
      },
      {
        id: 2,
        name: '李四',
        tel: '15088888888',
        addres: '北京市大兴京东大厦2号楼',
        isDefault: false,
        extends: [
          { label: '扩展1', value: '扩展信息展示' },
          { label: '扩展2', value: '扩展信息展示' }
        ]
      }
    ]
  };
  const clickEditHandel = jest.fn(i => i);
  const clickSelectHandel = jest.fn(i => i);
  const event = {
    onEdit: clickEditHandel,
    onSelected: clickSelectHandel
  }
  const { container } = render(
    <ReceiveInvoiceList list={state.list} defaultValue={state.defaultValue} onSelected={event.onSelected} onEdit={event.onEdit} />
  )
  const comp = container.querySelector('.nb-receive-invoice-list') as HTMLElement;
  let el = expect(comp);
  el.toBeTruthy();

  // 测试 edit 事件

  let testEidtIndex = 1;
  const editEl = container.querySelectorAll('.nut-icon-edit')[testEidtIndex];
  editEl && fireEvent.click(editEl);
  expect(clickEditHandel.mock.lastCall[0]).toBe(state.list[testEidtIndex]);

  // 测试 select 事件
  let testSelectIndex = 0;
  const selectEl = container.querySelectorAll('.nb-receive-invoice-list__item-header')[testSelectIndex];
  selectEl && fireEvent.click(selectEl);
  expect(clickSelectHandel.mock.lastCall[0]).toBe(state.list[testSelectIndex]);

  const selectFooterEl = container.querySelectorAll('.nb-receive-invoice-list__item-footer')[testSelectIndex];
  selectFooterEl && fireEvent.click(selectFooterEl);
  expect(clickSelectHandel.mock.lastCall[0]).toBe(state.list[testSelectIndex]);

  // 快照生成
  expect(comp).toMatchSnapshot();

})

test('test delete event', async () => {
  const state = {
    defaultValue: 1,
    list: [
      {
        id: 1,
        name: '张三',
        tel: '15088888888',
        addres: '北京市大兴京东大厦1号楼',
        isDefault: true,
      },
      {
        id: 2,
        name: '李四',
        tel: '15088888888',
        addres: '北京市大兴京东大厦2号楼',
        isDefault: false,
        extends: [
          { label: '扩展1', value: '扩展信息展示' },
          { label: '扩展2', value: '扩展信息展示' }
        ]
      }
    ]
  };
  const clickDeleteHandel = jest.fn(i => i);
  const event = {
    onDelete: clickDeleteHandel,
  }
  const { container } = render(
    <ReceiveInvoiceList enableDelete={true} list={state.list} defaultValue={state.defaultValue} onDelete={event.onDelete} />
  )

  // 测试 delete 事件

  let testDeleteIndex = 1;
  const delEl = container.querySelectorAll('.nut-button')[testDeleteIndex];
  delEl && fireEvent.click(delEl);
  expect(clickDeleteHandel.mock.lastCall[0]).toBe(state.list[testDeleteIndex]);


})
