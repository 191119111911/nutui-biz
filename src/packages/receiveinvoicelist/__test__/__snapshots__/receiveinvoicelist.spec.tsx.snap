// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test props && event 1`] = `
<div
  class="nb-receive-invoice-list"
  modelvalue="1"
>
  <div
    class="nut-cell-group nb-receive-invoice-list__item"
  >
    <div
      class="nut-cell-group__wrap"
    >
      <div
        class="nut-cell nb-receive-invoice-list__item-header "
        style="border-radius: 6px;"
        to=""
      >
        <div
          class="nut-cell__title"
        >
          <div
            class="nut-cell__maintitle"
          >
            <div
              class="nut-tag nut-tag--danger"
            >
              <span
                class="nut-tag-text"
              >
                默认
              </span>
            </div>
             
            张三
          </div>
        </div>
        <i
          class="nutui-iconfont nut-icon nut-icon-edit "
          style="color: rgb(102, 102, 102);"
        />
      </div>
      <div
        class="nut-cell nb-receive-invoice-list__item-footer "
        style="border-radius: 6px;"
        to=""
      >
        <div
          class="nut-checkbox "
        >
          <i
            class="nutui-iconfont nut-icon nut-icon-checked nut-checkbox__icon"
            style="font-size: 18px; height: 18px; width: 18px;"
          />
          <span
            class="nut-checkbox__label "
          />
        </div>
        <div
          class="nb-receive-invoice-list__item-footer infobox"
        >
          <div
            class="nb-receive-invoice-list__item-footer-info"
          >
            <span>
              电话
            </span>
            <p>
              15088888888
            </p>
          </div>
          <div
            class="nb-receive-invoice-list__item-footer-info"
          >
            <span>
              地址
            </span>
            <p>
              北京市大兴京东大厦1号楼
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="nut-cell-group nb-receive-invoice-list__item"
  >
    <div
      class="nut-cell-group__wrap"
    >
      <div
        class="nut-cell nb-receive-invoice-list__item-header "
        style="border-radius: 6px;"
        to=""
      >
        <div
          class="nut-cell__title"
        >
          <div
            class="nut-cell__maintitle"
          >
            李四
          </div>
        </div>
        <i
          class="nutui-iconfont nut-icon nut-icon-edit "
          style="color: rgb(102, 102, 102);"
        />
      </div>
      <div
        class="nut-cell nb-receive-invoice-list__item-footer "
        style="border-radius: 6px;"
        to=""
      >
        <div
          class="nut-checkbox "
        >
          <i
            class="nutui-iconfont nut-icon nut-icon-check-normal nut-checkbox__icon--unchecked"
            style="font-size: 18px; height: 18px; width: 18px;"
          />
          <span
            class="nut-checkbox__label "
          />
        </div>
        <div
          class="nb-receive-invoice-list__item-footer infobox"
        >
          <div
            class="nb-receive-invoice-list__item-footer-info"
          >
            <span>
              电话
            </span>
            <p>
              15088888888
            </p>
          </div>
          <div
            class="nb-receive-invoice-list__item-footer-info"
          >
            <span>
              地址
            </span>
            <p>
              北京市大兴京东大厦2号楼
            </p>
          </div>
          <div
            class="nb-receive-invoice-list__item-footer-info"
          >
            <span>
              扩展1
            </span>
            <p>
              扩展信息展示
            </p>
          </div>
          <div
            class="nb-receive-invoice-list__item-footer-info"
          >
            <span>
              扩展2
            </span>
            <p>
              扩展信息展示
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
